### PR TITLE
Implement duplicate account label check

### PR DESCRIPTION
**Pull request by we-ai**
This PR implements backend code for issue [#659](https://github.com/episphere/connect/issues/659), with some code cleanup. Below are details:
- Moved error handling to API level, for code simplicity
- Added duplicate flag check when verifying token and pin
- Moved common logic for token and pin checks to `verifyTokenOrPin`, to eliminate repeated code, based on comment from Abhinav
- Retired `verifyToken` and `verifyPin` functions, based on comment from Tony

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -17,40 +17,50 @@ const nciConceptId = `517700004`;
...
+    
+    if (participantData.state.uid === undefined) {
+      resultObj.isValidPin = true;
+      resultObj.docId = snapshot.docs[0].id;
     }
-}
+  }
+
+  return resultObj;
+};
```
Comment (abhinavjonnada82) [inline utils/firestore.js:63]: Suggestion:
`resultObj`, `if (snapshot.size === 1) {}` block can be changed to common function helper.

**Review by abhinavjonnada82 — APPROVED** [review]

**Comment by abhinavjonnada82** [comment]
Clean implementation.

**Comment by abhinavjonnada82** [comment]
Looks good with common refactor.

File: utils/validation.js [hunk 2]
```diff
@@ -45,51 +46,48 @@ const validateToken = async (req, res, uid) => {
...
+    }
 
-        const { verifyToken } = require('./firestore');
-        const isValid = await verifyToken(token);
+    if (token) {
+      const {
+        verifyToken,
+        linkParticipanttoFirebaseUID,
+      } = require('./firestore');
+      const { isDuplicateAccount, isValid: isValidToken, docId } = await verifyToken(token);
```
Comment (anthonypetersen) [inline utils/validation.js:60]: merging `verifyPIN()` and `verifyToken()` makes sense, but why keep the two functions just to setup to parameter we pass to `verifyTokenOrPin()`? Seems like an unnecessary middle man.

**Review by anthonypetersen — APPROVED** [review]
looks good

**Pull request merged**